test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/schema.py

```diff
@@ -143,6 +143,37 @@ def options_with_apply_defaults(v):
     return v
 
 
+def check_removal_version(v, version_name, collection_name_name, error_code='invalid-removal-version'):
+    version = v.get(version_name)
+    collection_name = v.get(collection_name_name)
+    if not isinstance(version, string_types) or not isinstance(collection_name, string_types):
+        # If they are not strings, schema validation will have already complained.
+        return v
+    if collection_name == 'ansible.builtin':
+        try:
+            parsed_version = StrictVersion()
+            parsed_version.parse(version)
+        except ValueError as exc:
+            raise _add_ansible_error_code(
+                Invalid('%s (%r) is not a valid ansible-base version: %s' % (version_name, version, exc)),
+                error_code=error_code)
+        return v
+    try:
+        parsed_version = SemanticVersion()
+        parsed_version.parse(version)
+        if parsed_version.major != 0 and (parsed_version.minor != 0 or parsed_version.patch != 0):
```

Comment:
```suggestion
        if parsed_version.major == 0 or (parsed_version.major != 0 and (parsed_version.minor != 0 or parsed_version.patch != 0)):
```

Reply:
Same as above, this is allowed by the spec.